src/pkgcheck/checks/pkgdir.py

```diff
@@ -252,13 +271,17 @@ def feed(self, pkgset):
                         yield EmptyFile(pjoin(base_dir, filename), pkg=pkg)
                     else:
                         files_by_size[file_stat.st_size].append(pjoin(base_dir, filename))
-                        if file_stat.st_size > 20480:
+                        total_size += file_stat.st_size
+                        if file_stat.st_size > SizeViolation.limit:
                             yield SizeViolation(
                                 pjoin(base_dir, filename), file_stat.st_size, pkg=pkg)
                     if banned_chars := set(filename) - allowed_filename_chars:
                         yield BannedCharacter(
                             pjoin(base_dir, filename), sorted(banned_chars), pkg=pkg)
 
+        if total_size > TotalSizeViolation.limit:
```

Comment:
FYI, I was considering something along the lines of:

```suggestion
        if sum(size*len(files) for size, files in files_by_size.items()) > TotalSizeViolation.limit:
```

but I've decided that plain `+=` should be both more readable and faster.

Reply:
I agree. Easier to read is more important in this place.